### PR TITLE
[feature/NRLF-XXXX-auth_from_s3] Auth from S3

### PR DESCRIPTION
**Pull request by jaklinger**

### CODE REVIEW

File: feature_tests/common/constants.py [hunk 1]
```diff
@@ -59,6 +60,9 @@
     DocumentPointer: DOCUMENT_POINTER_TABLE_DEFINITION,
 }
 
+PERMISSIONS_BUCKET = "permissions_bucket"  # This should match the value in TF outputs
```
Comment (Rohoolio) [inline feature_tests/common/constants.py:63]: Is this value right or does it get changed later? its name is now 

"${local.prefix}--authorization-store"

File: layer/lambda_utils/lambda_utils/header_config.py [hunk 2]
```diff
@@ -41,10 +41,13 @@ class ClientRpDetailsHeader(AbstractHeader):
 
 
 class ConnectionMetadata(AbstractHeader):
-    pointer_types: list[str] = Field(alias="nrl.pointer-types")
+    pointer_types: list[str] = Field(alias="nrl.pointer-types", default_factory=list)
     ods_code: str = Field(alias="nrl.ods-code")
     ods_code_extension: str = Field(alias="nrl.ods-code-extension", default=None)
-    nrl_permissions: list[str] = Field(alias="nrl.permissions", default=[])
+    nrl_permissions: list[str] = Field(alias="nrl.permissions", default_factory=list)
+    enable_permissions_lookup: bool = Field(
```
Comment (Rohoolio) [inline layer/lambda_utils/lambda_utils/header_config.py:48]: Im not sure the custom attribute can be a bool - will it be able to parse this?

File: layer/nrlf/nrlf/core/authoriser.py [hunk 3]
```diff
@@ -42,9 +54,12 @@ class Config(BaseModel):
     ENVIRONMENT: str
     SPLUNK_INDEX: str
     SOURCE: str
+    PERMISSIONS_LOOKUP_BUCKET: str
```
Comment (Rohoolio) [inline layer/nrlf/nrlf/core/authoriser.py:57]: I think we want to change this to be Authorization store to match the Terraform now

File: layer/nrlf/nrlf/core/authoriser.py [hunk 4]
```diff
@@ -58,6 +73,8 @@ def build_persistent_dependencies(config: Config) -> dict[str, any]:
         "environment": config.ENVIRONMENT,
         "splunk_index": config.SPLUNK_INDEX,
         "source": config.SOURCE,
+        "permissions_lookup_bucket": config.PERMISSIONS_LOOKUP_BUCKET,
```
Comment (Rohoolio) [inline layer/nrlf/nrlf/core/authoriser.py:76]: Same as above

File: layer/nrlf/nrlf/core/authoriser.py (hunk 3 above)
Comment (Rohoolio) [reply layer/nrlf/nrlf/core/authoriser.py:57]: Which I should have done when I committed my other PR haha

File: layer/nrlf/nrlf/core/tests/test_authoriser_utils.py [hunk 5]
```diff
@@ -79,6 +142,8 @@ def test_execute_steps_yields_deny_on_internal_server_error(
             ENVIRONMENT="an-environment",
             SPLUNK_INDEX="an-index",
             SOURCE="the-lambda-name",
+            S3_CLIENT="the-s3-client",
+            PERMISSIONS_LOOKUP_BUCKET="a-bucket",
```
Comment (Rohoolio) [inline layer/nrlf/nrlf/core/tests/test_authoriser_utils.py:146]: Same here

File: terraform/infrastructure/lambda.tf [hunk 6]
```diff
@@ -266,11 +266,13 @@ module "producer__authoriser_lambda" {
   layers      = [module.lambda-utils.layer_arn, module.nrlf.layer_arn, module.third_party.layer_arn]
   kms_key_id  = module.kms__cloudwatch.kms_arn
   environment_variables = {
-    PREFIX       = "${local.prefix}--"
-    ENVIRONMENT  = local.environment
-    SPLUNK_INDEX = module.firehose__processor.splunk.index
+    PREFIX                    = "${local.prefix}--"
+    ENVIRONMENT               = local.environment
+    SPLUNK_INDEX              = module.firehose__processor.splunk.index
+    PERMISSIONS_LOOKUP_BUCKET = aws_s3_bucket.authorization-store.id
```
Comment (Rohoolio) [inline terraform/infrastructure/lambda.tf:272]: This is where we want to change the env var to something more auth store related

File: terraform/infrastructure/lambda.tf [hunk 7]
```diff
@@ -289,9 +291,10 @@ module "consumer__authoriser_lambda" {
...
```
Comment (Rohoolio) [inline terraform/infrastructure/lambda.tf:297]: Nice catch - i did not remember to put the same env var in the consumer 😓

File: terraform/infrastructure/output.tf [hunk 8]
```diff
@@ -57,3 +57,7 @@ output "firehose" {
 output "mi" {
   value = module.mi
 }
+
+output "permissions_bucket" {
```
Comment (Rohoolio) [inline terraform/infrastructure/output.tf:61]: Again ive called this permissions bucket but we should probably adjust that to be consistent to authorization_store

File: feature_tests/common/constants.py (hunk 1 above)
Comment (jaklinger) [reply feature_tests/common/constants.py:63]: oh that comment's old actually, now it's just a dummy name for the local run (i.e. when mocking s3)

Comment (jaklinger) [reply feature_tests/common/constants.py:63]: so i'll rm the comment

File: layer/lambda_utils/lambda_utils/header_config.py (hunk 2 above)
Comment (jaklinger) [reply layer/lambda_utils/lambda_utils/header_config.py:48]: Seems to work fine:

image

Comment (jaklinger) [reply layer/lambda_utils/lambda_utils/header_config.py:48]: the value we pass to NRLF can certainly be a bool, more of a question about how we choose to parse this on the apigee side

File: layer/nrlf/nrlf/core/authoriser.py (hunk 3 above)
Comment (jaklinger) [reply layer/nrlf/nrlf/core/authoriser.py:57]: ah yep

**Pull request closed without merging**